pkg/cqrs/base_cqrs/execution.go

```diff
@@ -48,11 +52,39 @@ func (w wrapper) FunctionsByTrigger(ctx context.Context, eventName string) ([]in
 	all := []inngest.Function{}
 	for _, fn := range fns {
 		for _, t := range fn.Triggers {
-			if t.EventTrigger != nil && t.Event == eventName {
-				all = append(all, fn)
-				break
+			if t.EventTrigger != nil {
+				for _, trigger := range matchingTriggers {
+					if t.Event == trigger {
+						all = append(all, fn)
+						break
+					}
+				}
```

Comment:
This is not the most elegant, but for the low # of fns in dev server or self-host there isn't a need to overcomplicate.